underlay/build.gradle

```diff
@@ -1,50 +1,77 @@
 plugins {
     id 'java'
     id 'java-test-fixtures'
+    id 'pmd'
+
     id 'tanagra.java-conventions'
+    id 'org.springframework.boot' version '3.2.5'
+    id 'io.spring.dependency-management' version '1.1.5'
     id 'com.google.protobuf' version '0.9.4'
 }
 
-sourceCompatibility = JavaVersion.VERSION_11
+sourceCompatibility = JavaVersion.VERSION_17
 
 repositories {
     mavenCentral()
     maven {
-        url "https://broadinstitute.jfrog.io/broadinstitute/libs-release/"
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-release/'
     }
     maven {
-        url "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/"
+        url 'https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/'
     }
     maven {
-        url "https://packages.confluent.io/maven/"
+        url 'https://packages.confluent.io/maven/'
+    }
+}
+
+dependencyManagement {
+    imports {
+        mavenBom org.springframework.boot.gradle.plugin.SpringBootPlugin.BOM_COORDINATES
+        mavenBom 'io.opentelemetry:opentelemetry-bom:1.38.0'
+        mavenBom 'io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:2.4.0-alpha'
+    }
+    dependencies {
+        dependency "io.swagger.core.v3:swagger-annotations:${gradle.vSwaggerAnnotations}"
     }
 }
 
 dependencies {
-    implementation group: "bio.terra", name: "terra-common-lib", version: "0.0.69-SNAPSHOT"
-    implementation group: "org.apache.commons", name: "commons-text", version: "1.9"
-    testFixturesImplementation group: "org.slf4j", name: "slf4j-api", version: "1.7.25"
+    pmd "net.sourceforge.pmd:pmd-ant:${gradle.vPmd}"
+    pmdAux("io.swagger.core.v3:swagger-annotations:${gradle.vSwaggerAnnotations}")
+    implementation "io.swagger.core.v3:swagger-annotations:${gradle.vSwaggerAnnotations}"
 
-    // GCP libraries. Versions are controlled by the BOM specified in buildSrc.
-    implementation group: "com.google.cloud", name: "google-cloud-bigquery"
-    implementation group: "com.google.cloud", name: "google-cloud-storage"
-    implementation group: "com.google.guava", name: "guava"
+    implementation "com.fasterxml.jackson.core:jackson-core:${gradle.vJackson}"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${gradle.vJackson}"
+
+    implementation "org.apache.commons:commons-text:${gradle.vApacheCommonsText}"
+    testFixturesImplementation "org.slf4j:slf4j-api:${gradle.vSlf4jApi}"
+
+    // Spring version controlled dependencies
+    implementation 'org.springframework.retry:spring-retry:2.0.6'
```

Comment:
Comment says Spring controlled version -- should there be a version number here?

Reply:
fixed, see other related comment